allen-cell-animated/timelapse-colorizer
# feature: Toggle backdrop visibility

Problem
=======
Closes #480, "make background image toggle more visible"!

This change adds a new onscreen toggle for the backdrop image to help make it more discoverable for users. 

*Estimated review size: ~20 minutes, medium*

Solution
========
- Added a backdrop visible flag to the `ViewerConfig` type.
- Synced the new backdrop visibility flag to the URL.
- Changed object opacity so it is only applied when backdrops are visible.
- Updated `SettingsTab` with the new vector config.
  - This change moves object opacity to live under the backdrop settings area.
- Adds a new onscreen toggle for the background image in `CanvasWrapper`.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-483/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fassay-dev%2Fusers%2FChantelle%2Fcolorizer%2Fnew_backdrop%2Fexploratory_dataset%2Fcollection.json
2. Hover over the backdrop button on the UI. Try clicking the link to go to the viewer settings. Toggle the button.
4. Open a preview link where backdrops are not included: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-483/viewer?collection=https%3A%2F%2Fallencell.s3.amazonaws.com%2Faics%2Fnuc-morph-dataset%2Ftimelapse_feature_explorer_datasets%2Fexploratory_dataset%2Fcollection.json
5. Observe the disabled state.

Screenshots (optional):
-----------------------
**Video Talk-through (🔊):**

https://github.com/user-attachments/assets/af1d9cb6-9438-43be-a7fa-b0265cc4feea


![image](https://github.com/user-attachments/assets/3d410eee-d04d-4983-a9c1-241099b4a4fb)

![image](https://github.com/user-attachments/assets/70b03537-1ccf-4b43-b392-a9d752db983c)

![image](https://github.com/user-attachments/assets/6ba06040-e38e-45e1-b363-5c853489d18a)
